json-schema-generator/src/main/java/io/micronaut/jsonschema/generator/utils/GeneratorContext.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jsonschema.generator.utils;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.jsonschema.generator.aggregator.AnnotationsAggregator;
+import io.micronaut.jsonschema.model.Schema;
+import io.micronaut.sourcegen.model.ClassTypeDef;
+import io.micronaut.sourcegen.model.TypeDef;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static io.micronaut.jsonschema.generator.SourceGenerator.getInputFileName;
+import static io.micronaut.jsonschema.generator.aggregator.TypeAggregator.getClassName;
+import static io.micronaut.jsonschema.generator.aggregator.TypeAggregator.getTypeDefFromJson;
+import static io.micronaut.jsonschema.model.Schema.DEF_SCHEMA_REF_PREFIX;
+import static io.micronaut.jsonschema.model.Schema.ONE_OF_SCHEMA_REF_PREFIX;
+
+/**
+ * An aggregator for storing and accessing definitions and oneOf relations from json schema.
+ * Saves a map of definition reference to TypeDef, Boolean (boolean value is true if the type is a class/interface object).
+ * Saves a map of oneOf objects to keep in track on inheriting objects.
+ *
+ * @author Elif Kurtay
+ * @since 1.3
+ */
+@Internal
+public final class GeneratorContext {
+    private final HashMap<String, Map.Entry<TypeDef, Boolean>> DEFINITIONS = new HashMap<>();
+    private final HashMap<String, LinkedList<String>> TEMP_DEFINITIONS = new HashMap<>();
+    private final HashMap<String, Schema> ONE_OF_SET = new HashMap<>();
```

Comment:
does the ordering of these maps matter? Will it change the order things are written out?

Reply:
not really